src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -297,15 +299,18 @@ internal bool ReportUseSiteDiagnostic(DiagnosticInfo? info, Location location)
 
     internal readonly struct ImmutableBindingDiagnostic<TAssemblySymbol> where TAssemblySymbol : class, IAssemblySymbolInternal
     {
-        public readonly ImmutableArray<Diagnostic> Diagnostics;
-        public readonly ImmutableArray<TAssemblySymbol> Dependencies;
+        private readonly ImmutableArray<Diagnostic> _diagnostics;
+        private readonly ImmutableArray<TAssemblySymbol> _dependencies;
+
+        public ImmutableArray<Diagnostic> Diagnostics => _diagnostics.NullToEmpty();
+        public ImmutableArray<TAssemblySymbol> Dependencies => _dependencies.NullToEmpty();
 
         public static ImmutableBindingDiagnostic<TAssemblySymbol> Empty => new ImmutableBindingDiagnostic<TAssemblySymbol>(default, default);
 
         public ImmutableBindingDiagnostic(ImmutableArray<Diagnostic> diagnostics, ImmutableArray<TAssemblySymbol> dependencies)
         {
-            Diagnostics = diagnostics.NullToEmpty();
-            Dependencies = dependencies.NullToEmpty();
+            _diagnostics = diagnostics.NullToEmpty();
```

Comment:
>.NullToEmpty() [](start = 38, length = 14)

nit: Do we need the `.NullToEmpty()` here, since already done when returning from properties above?

Reply:
> Do we need the .NullToEmpty() here, since already done when returning from properties above?

I don't think it hurts.

---
In reply to: [372085003](https://github.com/dotnet/roslyn/pull/41268#discussion_r372085003) [](ancestors = 372085003)